src/ApparentHorizons/YlmSpherepack.hpp

```diff
@@ -0,0 +1,231 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#pragma once
+
+#include <cstddef>
+#include <vector>
+
+#include "DataStructures/DataVector.hpp"
+#include "DataStructures/Tensor/Tensor.hpp"
+#include "YlmSpherepackStorage.hpp"
+
+/// \ingroup Spectral
+/// \brief
+/// C++ interface to SPHEREPACK.
+class YlmSpherepack {
+ public:
+  using first_deriv_t = tnsr::i<DataVector, 2, Frame::Logical>;
+  using second_deriv_t = tnsr::ij<DataVector, 2, Frame::Logical>;
+  using interpolation_info_t = std::vector<std::vector<double>>;
+
+  /// Here l_max and m_max are the largest fully-represented l and m in
+  /// the Ylm expansion.
+  YlmSpherepack(size_t l_max, size_t m_max) noexcept;
+
+  bool operator==(const YlmSpherepack& rhs) const noexcept;
+
+  /// Static functions to return the correct sizes of vectors of
+  /// collocation points and spectral coefficients for a given l_max
+  /// and m_max.  Useful for allocating space without having to create
+  /// a YlmSpherepack.
+  static size_t physical_size(const size_t l_max, const size_t m_max) noexcept {
+    return (l_max + 1) * (2 * m_max + 1);
+  }
+  static size_t spectral_size(const size_t l_max, const size_t m_max) noexcept {
+    return (2 * (l_max + 1) * (m_max + 1));
+  }
+
+  /// Sizes in physical and spectral space for this instance.
```

Comment:
This comment and the four functions should be wrapped with //@{ //@} to group them in doxygen.